### ISSUE TITLE
warnings that should be ignored sometiems suppress legitimate warnings

### ISSUE THREAD

**Issue by kaizhu256** [jslint-org/jslint#408]
**Describe the bug**
1. A clear and concise description of what the bug is.

jslint directive `... //jslint-ignore-line

2. Jslint edition:
v2022.7.1-beta

3. Code snippet (if applicable):

```javascript
foo(); // bug - legitimate warning "Undeclared 'foo'." is suppressed
function bar() {
    let baz = 0;
    baz.aa = baz.bb = function () { //jslint-ignore-line
        return;
    };
}
```

**Expected behavior**

in above code-snippet, jslint should warn `Undeclared 'foo'.` but the warning was suppressed by directive `//jslint-ignore-line`
